elide-core/src/main/java/com/yahoo/elide/core/EntityDictionary.java

```diff
@@ -872,7 +889,29 @@ public void bindEntity(Class<?> cls) {
         }
 
         bindJsonApiToEntity.put(type, declaredClass);
-        entityBindings.put(declaredClass, new EntityBinding(this, declaredClass, type, name));
+        entityBindings.put(declaredClass, new EntityBinding(this, declaredClass, type, name, hiddenAnnotations));
+        if (include.rootLevel()) {
+            bindEntityRoots.add(declaredClass);
+        }
+    }
+
+    /**
+     * Add an EntityBinding instance to dictionary.
+     *
+     * @param entityBinding EntityBinding instance
+     */
+    public void bindEntity(EntityBinding entityBinding) {
+        Class<?> declaredClass = entityBinding.entityClass;
```

Comment:
Why not have this method call the new bindEntity method with an empty set (or a set that includes Exclude)?

Reply:
This `bindEntity` method needs to keep all information from the existing binding, as the `hiddenAnnotations` are not tracked in `EntityBinding`, if we bind the class again in a different dictionary, previous `hiddenAnnotations` would not be applied.